### DISCUSSION TITLE
Avoid render after listen a global event

### DISCUSSION THREAD

**Discussion by itepifanio** [livewire/livewire#1026] [4. General]
There's a way to avoid render the component after listen a global event?

In my blade I deal with checkboxes with JS, when the users clicks at "Mark As View" I emit an event with the ids selected by the user:

```js
window.livewire.emit('MarkAsRead', ids); 
```

And in the component would be just a:

```php
protected $listeners = ['MarkAsRead'    => 'markAsRead',];
...
public function markAsRead(){
    Model::whereIn('id', $ids)->update(['view' => true]);
}
...
```

My Livewire component uses a tricky query and a private variable that are been passed in the render, so I would like to avoid render again (the changes I made in the DOM).
I'm not currently using the `wire:model` in my checkboxes because the way that the pages and components are been structured the global event makes more sense.

**Comment by mblarsen** [comment]
> I'm not currently using the `wire:model` in my checkboxes because the way that the pages and components are been structured the global event makes more sense.

Could you show your code for this? It is a bit hard to understand what the problem really is.

**Reply by itepifanio** [reply]
Sure, I have this blade:
```php
@section
...
<div class="flex-1 mx-auto mb-20 lg:mb-0" x-data="markAsRead">
...
<div class="flex w-full mb-4">
            <div class="relative flex-1 mr-6 my-2">
                <div class="inline-flex items-center px-8 py-3 bg-white rounded-lg">
                    <input type="checkbox" class="form-checkbox h-5 w-5" x-on:change="checkAll($event)">
                    <div class="inline-flex items-center ml-4 cursor-pointer text-gray-650"
                         x-on:click="livewireMarkAsRead">
                        @svg('circle-checked-outline', 'w-6 h-6 mx-3')
                        <span class="text-md text-gray-650 underline font-medium">Mark as Read</span>
                    </div>
                    <div class="hidden inline-flex items-center ml-4 cursor-pointer text-gray-650">
                        @svg('circle-outline', 'w-6 h-6 mx-3')
                        <span class="text-md text-gray-650 underline font-medium">Assign</span>
                    </div>
                </div>
            </div>
        </div>
</div>
...
@livewire('overview.overview-list')
....
@endsection


@push('scripts')
    <script>
        var markAsRead = {
            // Get checkboxes selected and send to livewire
            livewireMarkAsRead() {
                let textinputs = document.querySelectorAll('input[type=checkbox]');

                let ids = [...textinputs].reduce((acc, item) => {
                    if(!isNaN(item.value) && item.checked){
                        acc.push(item.value);
                    }
                    return acc;
                }, []);

                window.livewire.emit('MarkAsRead', ids);
                // changes the dom to not rerender
                ids.forEach(function (elem) {
                    let isRead = document.getElementsByClassName(`isRead[${elem}]`);
                    isRead[0].style.display = '';
                    isRead[1].style.display = 'none';
                })
            },
            checkAll(e){
                let textinputs = document.querySelectorAll('input[type=checkbox]');
                textinputs.forEach(function (elem) {
                    elem.checked = e.target.checked;
                });
            }
        }
    </script>
@endpush
```

This page is not a livewire component, but I send the event globally and its listened by the overview-list-component

```php
class OverviewList extends Component
{
   protected $listeners = [
        ...
        'MarkAsRead'    => 'markAsRead',
    ];

    public function mount()
    {
        $this->templates = $this->loadTemplates();
    }

    public function render()
    {
        return view('overview.overview-list', [
            'templates'  => $this->loadTemplates(),
        ]);
    }

    public function loadTemplates()
    {
        return TemplateService::all();
    }

    public function markAsRead($messagesIds)
    {
        Chat::whereIn('id', $messagesIds)->update([
            'read' => true,
        ]);
    }
}
```
I would like to call the markAsRead function, but not render all data, like an ajax call, it's possible?

**Reply by mblarsen** [reply]
One approach could be to make the check list items Livewire components. Livewire is different from other frameworks that uses nested components in that updates to the parent doesn't re-render the child components automatically. 

You can pass in the necessary props needed to render the item to avoid having the list item component do any DB queries.

Alternatively you can try to use `wire:ignore` or `wire:ignore.self`. Livewire will only render the first time, but not on updates. Then it is up to you to update the DOM with Alpine.js or what not.

**Reply by itepifanio** [reply]
` wire:ignore ` isn't an option because other interactions update the children and need rerender. I checked out the computed properties at the docs but it looks like this doesn't work with events. I will pass the props needed, appreciate the help, thanks.
